hadoop-tools/hadoop-azure/src/test/java/org/apache/hadoop/fs/azurebfs/services/TestGetContentSummary.java

```diff
@@ -0,0 +1,204 @@
+      "/testFolder/testFolder2/testFolder5", "/testFolder/testFolder3"};
+
+  private final AzureBlobFileSystem fs = createFileSystem();
+  private final int testBufferSize = 20;
+  private final int filesPerDirectory = 2;
+  private final int numFilesForListMaxTest = DEFAULT_AZURE_LIST_MAX_RESULTS + 10;
+  private final byte[] b = new byte[testBufferSize];
+  private final int maxThreads = 16;
+
+  public TestGetContentSummary() throws Exception {
+    createDirectoryStructure();
+    new Random().nextBytes(b);
+  }
+
+  @Test
+  public void testFilesystemRoot()
+      throws IOException {
+    int fileCount =
+        (directories.length - 2) * filesPerDirectory + numFilesForListMaxTest;
+    ContentSummary contentSummary = fs.getContentSummary(new Path("/"));
+    checkContentSummary(contentSummary, directories.length, fileCount,
+        dirsWithNonEmptyFiles.length * filesPerDirectory * testBufferSize);
+  }
+
+  @Test
+  public void testFileContentSummary() throws IOException {
+    ContentSummary contentSummary = fs.getContentSummary(pathToFile);
+    checkContentSummary(contentSummary, 0, 1, testBufferSize);
+  }
+
+  @Test
+  public void testLeafDir() throws IOException {
+    ContentSummary contentSummary = fs.getContentSummary(pathToLeafDir);
+    checkContentSummary(contentSummary, 0, 0, 0);
+  }
+
+  @Test
+  public void testIntermediateDirWithFilesOnly() throws IOException {
+    ContentSummary contentSummary =
+        fs.getContentSummary(pathToIntermediateDirWithFilesOnly);
+    checkContentSummary(contentSummary, 0, filesPerDirectory,
+        testBufferSize * filesPerDirectory);
+  }
+
+  @Test
+  public void testIntermediateDirWithFilesAndSubdirs() throws IOException {
+    ContentSummary contentSummary =
+        fs.getContentSummary(pathToIntermediateDirWithFilesAndSubdirs);
+    checkContentSummary(contentSummary, 3, 3 * filesPerDirectory,
+        testBufferSize * filesPerDirectory);
+  }
+
+  @Test
+  public void testDirOverListMaxResultsItems()
+      throws IOException {
+    checkContentSummary(
+        fs.getContentSummary(pathToListMaxDir), 1,
+        numFilesForListMaxTest + filesPerDirectory, 0);
+  }
+
+  @Test
+  public void testInvalidPath() throws Exception {
+    intercept(IOException.class, () -> fs.getContentSummary(new Path(
+        "/nonExistentPath")));
+  }
+
+  @Test
+  public void testConcurrentGetContentSummaryCalls()
+          throws InterruptedException, ExecutionException {
+    ExecutorService executorService = new ThreadPoolExecutor(1,
+            maxThreads, 5, TimeUnit.SECONDS, new SynchronousQueue<>());
+    ArrayList<Future<ContentSummary>> futures = new ArrayList<>();
+    for (String directory : directories) {
+      Future<ContentSummary> future = executorService.submit(
+              () -> fs.getContentSummary(new Path(directory)));
+      futures.add(future);
+    }
+    int[][] dirCS = {{8, 8 * filesPerDirectory, 8 * testBufferSize}, {0, filesPerDirectory, 2 * testBufferSize},
```

Comment:
move each array of elements into separate lines for better readability

Reply:
done